xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp

```diff
@@ -2056,15 +2055,15 @@ void CLinuxRendererGL::DeleteYV12Texture(int index)
     return;
 
   /* finish up all textures, and delete them */
-  for(int f = 0;f<MAX_FIELDS;f++)
+  for(auto& field : m_buffers[index].fields)
   {
-    for(int p = 0;p<YuvImage::MAX_PLANES;p++)
+    for(CYuvPlane& p : field)
     {
-      if (m_buffers[index].fields[f][p].id)
+      if (p.id)
       {
-        if (glIsTexture(m_buffers[index].fields[f][p].id))
-          glDeleteTextures(1, &m_buffers[index].fields[f][p].id);
-        m_buffers[index].fields[f][p].id = 0;
+        if (glIsTexture(p.id))
+          glDeleteTextures(1, &p.id);
+        p.id = 0;
```

Comment:
it is important to be aware of field and plane index. there are still places left where range based loops won't fit, i.e.

```
  for (int f = FIELD_FULL; f<=FIELD_BOT ; f++)
  {
    int fieldshift = (f==FIELD_FULL) ? 0 : 1;
```

mixing two concepts make the code harder to read

Reply:
Where should I revert it back to the current style, only here, the whole file or additionally also `LinuxRendererGLES.cpp`?

Reply:
VideoRenderers/* seems easiest to me